src/brsTypes/components/RoSGNode.ts

```diff
@@ -1109,6 +1109,8 @@ export function createNodeByType(interpreter: Interpreter, type: BrsString) {
             }
 
             interpreter.inSubEnv(subInterpreter => {
+                let mPointer = subInterpreter.environment.getM();
+                mPointer.set(new BrsString("top"), node);
```

Comment:
Will this work for the base `RoSGNode`? Seems like this is only being added for custom components

Reply:
I thought it only had to be added for custom components. I am unsure if base RoSGNode uses `m.top` at all?

Reply:
Think I didn't explain myself correctly but I was talking about the base `Node` type. At the very top of this function there is an early return if the type is `Node`, thus basic nodes wouldn't have a `m.top` reference, right?

Reply:
Ahh I see. Is there a use case for that as well? Do base `Node` types ever have scripts associated with them?

Reply:
I don't think so.

Reply:
Yeah it should be fine to not apply this kind of override to the generic `Node` type — the only time we'd need access to a `Node`'s interface is from a JavaScript function, at which point we shouldn't need to route through `someInterpreter.environment.getM().get("top")` (since those public fields are already available through `this`)